Formula/koka.rb

```diff
@@ -0,0 +1,53 @@
+class Koka < Formula
+  desc "Compiler for the Koka language"
+  homepage "http://koka-lang.org"
+  url "https://github.com/koka-lang/koka/releases/download/v2.3.8/koka-v2.3.9-source.tar.gz"
+  sha256 "ae3e8127b46e51cc4df3525c5f913ecfbaeeb29f0a268e4a3d2e0ac809bc987f"
+  license "Apache-2.0"
+  head "https://github.com/koka-lang/koka.git", branch: "dev"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
+  depends_on "vcpkg"
+
+  def install
+    system "cabal", "new-update"
+    system "cabal", "new-build", "-j"
+    system "cabal", "new-run", "koka", "--",
+           "-e", "util/bundle.kk", "--",
+           "--prefix=bundle/local-brew", "--postfix=brew"
+    share.install Dir["bundle/local-brew/share/*"]
+    lib.install Dir["bundle/local-brew/lib/*"]
+    bin.install Dir["bundle/local-brew/bin/*"]
+  end
+
+  def caveats
+    if File.exist?("/usr/local/bin/koka")
+      kokaver = `/usr/local/bin/koka --version --console=raw`
+      kokaver = kokaver[/Koka\s*v?([\d.]+)/, 1]
+      <<~EOS
+        It seems you have already koka v#{kokaver} installed at '/usr/local/bin/koka'.
+        This version may hide the brew installed koka as it may come earlier in the PATH.
+        You can uninstall the previous koka fully using:
+
+        $ curl -sSL https://github.com/koka-lang/koka/releases/latest/download/install.sh | sh -s -- --uninstall --version=v#{kokaver}
+
+        or by only removing the symlink (to '/usr/local/bin/koka-v#{kokaver}'):
+
+        $ sudo rm /usr/local/bin/koka
+      EOS
+    else
+      ""
+    end
+  end
```

Comment:
This isn't really specific to Homebrew or they way we package something, which is what `caveats` are for. Let's leave this out.

Reply:
I agree; but if a user had already installed koka before, and now starts using `brew`, then the new install will not work as the older koka will be found first.  This message will make this apparent and potentially save people from trying to figure out why the install is not working. Hope we can leave it in?

Reply:
The installation will actually fail at the link stage if `/usr/local/bin/koka` already exists (Homebrew will not overwrite the file if it is already there). This message would not be displayed; Homebrew would display an error message instead.

Reply:
ah I see; on my M1 brew installs to `/opt/homebrew/bin` but I see that is not always the case. I guess it's best to remove it. I'll add a note on the Koka install page instead.

Reply:
Just to clarify, users running Intel Macs would see the Homebrew error message since Homebrew would try to install to `/usr/local/bin/koka`; on M1 Macs Homebrew installs to `/opt/homebrew/bin/koka` so there would be no error message/conflict.